# __TelecomX - Parte 2: Modelo de prediccion de cancelaciones__

# Objetivos

- Preparar los datos para el modelado (tratamiento, codificación, normalización).

- Realizar análisis de correlación y selección de variables.

- Entrenar dos o más modelos de clasificación.

- Evaluar el rendimiento de los modelos con métricas.

- Interpretar los resultados, incluyendo la importancia de las variables.

- Crear una conclusión estratégica señalando los principales factores que influyen en la cancelación.

# Entregables

- Analisis completo de los datos.

- Exportar modelo final.

- Informe con los hallazgos y recomendaciones.

# Preparacion de los datos

## Carga y exploracion

In [1]:
# libreria
import pandas as pd

In [2]:
# cargar datos previamente tratados desde github
data = pd.read_csv('https://raw.githubusercontent.com/jbernalg/challenge_TelecomX/refs/heads/main/data_out/datos_tratados.csv', sep=',')
data.head()

cancelacion     genero jubilado en_pareja dependientes  meses_contrato  \
0            0   Femenino       No        Si           Si               9   
1            0  Masculino       No        No           No               9   
2            1  Masculino       No        No           No               4   
3            1  Masculino       Si        Si           No              13   
4            1   Femenino       Si        Si           No               3   

  servicio_telefonico multiples_lineas servicio_internet seguridad_online  \
0                  Si               No               DSL               No   
1                  Si               Si               DSL               No   
2                  Si               No      Fibra_optica               No   
3                  Si               No      Fibra_optica               No   
4                  Si               No      Fibra_optica               No   

   ... proteccion_dispositivo soporte_efectivo streaming_TV  \
0  ...                     No               Si           Si   
1  ...                     No               No           No   
2  ...                     Si               No           No   
3  ...                     Si               No           Si   
4  ...                     No               Si           Si   

  streaming_peliculas tipo_contrato facturacion_electronica  \
0                  No       un year                      Si   
1                  Si     mes a mes                      No   
2                  No     mes a mes                      Si   
3                  Si     mes a mes                      Si   
4                  No     mes a mes                      Si   

          metodo_pago cuenta_mensual  cuentas_total  cuentas_diarias  
0   cheque por correo           65.6         593.30         0.242963  
1   cheque por correo           59.9         542.40         0.221852  
2  cheque electronico           73.9         280.85         0.615833  
3  cheque electronico           98.0        1237.85         0.251282  
4   cheque por correo           83.9         267.40         0.932222  

[5 rows x 21 columns]

In [3]:
# informacion de cada variable
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7032 entries, 0 to 7031
Data columns (total 21 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   cancelacion              7032 non-null   int64  
 1   genero                   7032 non-null   object 
 2   jubilado                 7032 non-null   object 
 3   en_pareja                7032 non-null   object 
 4   dependientes             7032 non-null   object 
 5   meses_contrato           7032 non-null   int64  
 6   servicio_telefonico      7032 non-null   object 
 7   multiples_lineas         7032 non-null   object 
 8   servicio_internet        7032 non-null   object 
 9   seguridad_online         7032 non-null   object 
 10  respaldo_online          7032 non-null   object 
 11  proteccion_dispositivo   7032 non-null   object 
 12  soporte_efectivo         7032 non-null   object 
 13  streaming_TV             7032 non-null   object 
 14  streaming_peliculas     

In [5]:
# distribucion de cancelacion
data['cancelacion'].value_counts(normalize=True)*100

cancelacion
0    73.421502
1    26.578498
Name: proportion, dtype: float64

La proporcion de la variable objetivo es la siguiente:

> 73.42% de los clientes se mantienen en la empresa

> 26.57% de los clientes abandonan la empresa

> Se evidencia un desequilibrio entre los clientes que permanecen y abandonan la empresa. Debe ser tratado.

## Limpiar los datos